utils/posix.h

```diff
@@ -0,0 +1,47 @@
+#ifndef POSIX_H
+#define POSIX_H
+
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+
+#ifdef HAVE_POSIX_SHM
+# define UFTRACE_SHMDIR_NAME "/dev/shm"
+#else
+# define UFTRACE_SHMDIR_NAME "uftrace.data/shm"
+#endif
+
+#ifndef HAVE_POSIX_SHM
+static inline int shm_open(const char *file, int oflag, mode_t mode)
+{
+	int ret;
+	char* path;
+	if(mkdir(UFTRACE_SHMDIR_NAME, S_IRWXU | S_IRWXG | S_IRWXO) && errno != EEXIST)
+		return -1;
+	const size_t path_size = sizeof(UFTRACE_SHMDIR_NAME) - 1 + strlen(file) + 1;
+	path = calloc(path_size, sizeof(char));
+	strncat(path, UFTRACE_SHMDIR_NAME, path_size - strlen(path) - 1);
+	strncat(path, file, path_size - strlen(path) - 1);
+	ret = open(path, oflag, mode);
```

Comment:
And I think we should fallback to other shared memory (if exists) instead of a regular file.  Maybe Sys-V shmem or ashmem (on Android) or just a regular file in a tmpfs.

Reply:
There is a problem.
Sys-V shmem and ashmem accepts size of the shared memory as a argument.

Reply:
Yeah you're right